packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -138,19 +138,21 @@ const processNodeUrl = (
     }
 
     if (
-      (url[0] === '/' && url[1] !== '/') ||
-      // #3230 if some request url (localhost:3000/a/b) return to fallback html, the relative assets
-      // path will add `/a/` prefix, it will caused 404.
-      //
-      // skip if url contains `:` as it implies a url protocol or Windows path that we don't want to replace.
-      //
-      // rewrite `./index.js` -> `localhost:5173/a/index.js`.
-      // rewrite `../index.js` -> `localhost:5173/index.js`.
-      // rewrite `relative/index.js` -> `localhost:5173/a/relative/index.js`.
-      ((url[0] === '.' || isBareRelative(url)) &&
-        originalUrl &&
-        originalUrl !== '/' &&
-        htmlPath === '/index.html')
+      // avoid duplicate base prefix when applying html transforms after assets plugin
+      !url.startsWith(config.base) &&
```

Comment:
We can't do this, or we won't support projects having `/root-path/base/` in the fs.
`/base/base/file.js` is a valid module.